runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -97,6 +97,33 @@ local function tags_lsp_to_vim(diagnostic, client_id)
   return tags
 end
 
+---@param related_informations lsp.DiagnosticRelatedInformation[]
+---@param offset_encoding string
+---@return vim.DiagnosticRelatedInformation[]
+local function related_information_lsp_to_vim(related_informations, offset_encoding)
+  return vim.iter(related_informations):map(
+  --- @param related_information lsp.DiagnosticRelatedInformation
+  --- @return vim.DiagnosticRelatedInformation
+    function(related_information)
+      local fname = vim.uri_to_fname(related_information.location.uri)
+      local bufnr = vim.fn.bufadd(fname)
+      if not bufnr then
+        return
+      end
+      local buf_lines = get_buf_lines(bufnr)
+      local start = related_information.location.range.start
+      local _end = related_information.location.range['end']
+      --- @type vim.DiagnosticRelatedInformation
+      return {
+        lnum = start.line,
+        col = line_byte_from_position(buf_lines, start.line, start.character, offset_encoding),
+        end_lnum = _end.line,
+        end_col = line_byte_from_position(buf_lines, _end.line, _end.character, offset_encoding),
+        message = related_information.message,
+      }
+    end)
```

Comment:
You also need `:totable()` to collapse the iterator back into a table (analagous to `collect()` in Rust)

```suggestion
    end):totable()
```